src/wallet/wallet.cpp

```diff
@@ -2285,10 +2311,10 @@ bool CWallet::GetCoinsToSpend(
     uint64_t best_spend_val = val;
 
     int minimum = INT_MAX - 1;
-    while(index >= required) {
-        int temp_min = (next_row[index] + GetRequiredCoinCountForAmount(
-            (index - required) * zeros, denominations));
-        if (minimum > temp_min && next_row[index] != (INT_MAX - 1) / 2) {
+    while(index >= roundedRequired) {
+        int temp_min = next_row[index] + GetRequiredCoinCountForAmount(
+            (index - roundedRequired) * zeros, denominations);
+        if (minimum > temp_min && next_row[index] != (INT_MAX - 1) / 2 && temp_min <= coinsLimit) {
```

Comment:
I think this check must be next_row[index] <= coinsLimit, because mints count must not matter, we must count only required spends.

Reply:
nice catch!